mypy/semanal.py

```diff
@@ -2355,6 +2356,41 @@ def is_classvar(self, typ: Type) -> bool:
     def fail_invalid_classvar(self, context: Context) -> None:
         self.fail('ClassVar can only be used for assignments in class body', context)
 
+    def process_module_assignment(self, s: AssignmentStmt) -> None:
+        """Check if s assigns a module an alias name; if so, update symbol table."""
+        # TODO support module alias assignment not in global scope
+        if not self.is_module_scope():
+            return
+        self._process_module_assignment(s.lvalues, s.rvalue, s)
+
+    def _process_module_assignment(
+            self,
+            lvals: List[Expression],
+            rval: Expression,
+            ctx: AssignmentStmt,
+    ) -> None:
+        """Propagate module references across assignments.
+
+        Recursively handles the simple form of iterable unpacking; doesn't
+        handle advanced unpacking with *rest, dictionary unpacking, etc.
+
+        """
+        if all(isinstance(v, (TupleExpr, ListExpr)) for v in lvals + [rval]):
+            litemlists = [v.items for v in cast(List[Union[TupleExpr, ListExpr]], lvals)]
+            ritems = cast(Union[TupleExpr, ListExpr], rval).items
+            for rv, *lvs in zip(ritems, *litemlists):
+                self._process_module_assignment(lvs, rv, ctx)
```

Comment:
Ok! Pushed an update that breaks down the code a bit and adds comments to clarify the intent. Let me know if that helps.